pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -65,119 +77,208 @@ func NewIngressAnalyzerWithObjects(objects []scan.K8sObject, pe *eval.PolicyEngi
+)
+
+// mapk8sIngressToServices populates k8sIngressToServicesMap
+func (ia *IngressAnalyzer) mapk8sIngressToServices(ing *netv1.Ingress) {
+	services := ia.getk8sIngressServices(ing)
+	if len(services) == 0 { // all ingress backends were ignored
+		return
+	}
+	if _, ok := ia.k8sIngressToServicesMap[ing.Namespace]; !ok {
+		ia.k8sIngressToServicesMap[ing.Namespace] = make(map[string][]serviceInfo)
+	}
+	ia.k8sIngressToServicesMap[ing.Namespace][ing.Name] = services
+}
+
+// getk8sIngressServices given k8s-Ingress object returns its targeted services info
+func (ia *IngressAnalyzer) getk8sIngressServices(ing *netv1.Ingress) []serviceInfo {
+	ingressStr := types.NamespacedName{Namespace: ing.Namespace, Name: ing.Name}.String()
+	backendServices := make([]serviceInfo, 0)
+	// if DefaultBackend is provided add its service info to the result
+	if ing.Spec.DefaultBackend != nil {
+		if ing.Spec.DefaultBackend.Service == nil {
+			ia.logger.Warnf(scan.Ingress + " " + ingressStr + ": " + defaultBackendWarning)
+		} else {
+			backendServices = append(backendServices, getServiceInfo(ing.Spec.DefaultBackend.Service))
+		}
+	}
+	// add service names from the Ingress rules
+	for _, rule := range ing.Spec.Rules {
+		for _, path := range rule.IngressRuleValue.HTTP.Paths {
+			if path.Backend.Service == nil {
+				ia.logger.Warnf(scan.Ingress + " " + ingressStr + ": " + ruleBackendWarning)
+			} else {
+				backendServices = append(backendServices, getServiceInfo(path.Backend.Service))
+			}
+		}
+	}
+	return backendServices
+}
+
+// utility func
+// getServiceInfo returns serviceInfo struct (name and port) from a given Ingress backend service
+func getServiceInfo(backendService *netv1.IngressServiceBackend) serviceInfo {
+	res := serviceInfo{serviceName: backendService.Name}
+	// Port.Name and Port.Number are mutually exclusive
+	if backendService.Port.Name != "" {
+		res.serviceTargetPort.StrVal = backendService.Port.Name
+	} else {
+		res.serviceTargetPort.IntVal = backendService.Port.Number
+	}
+	return res
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////
 // Ingress allowed connections
 
 // AllowedIngressConnections returns a map of the possible connections from ingress-controller pod to workload peers,
 // as inferred from Ingress and Route resources. The map is from a workload name to its connection object.
-func (ia *IngressAnalyzer) AllowedIngressConnections() map[string]eval.Connection {
+func (ia *IngressAnalyzer) AllowedIngressConnections() (map[string]common.Connection, error) {
 	// if there is at least one route/ ingress object that targets a service which selects a dst peer,
 	// then we have ingress connections to that peer
 
 	// get all targeted workload peers and compute allowed conns of each workload peer
-	res := make(map[string]eval.Connection)
-	for ns, rtSvcMap := range ia.routesToServicesMap {
+	// 1. from routes
+	routesResult := ia.allowedIngressConnectionsByRoutes()
```

Comment:
in the routes analysis, why ignoring `targetPort` specified in the resource?

Reply:
will add it (at first we preferred to not include it)

Reply:
I see in the examples that the route.targetPort may refer to the service.port.name which may differ from the service.port.targetPort(which eventually is the port in the peer)

so I'll change the servicesToPeersMap from being services -> peers to be services -> {peers , servicePorts}.